components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -403,56 +402,50 @@ func (g *rtkSerial) receiveAndWriteSerial() {
 		scanner = rtcm3.NewScanner(g.reader)
 	}
 
-	g.mu.Lock()
-	g.isConnectedToNtrip = true
-	g.mu.Unlock()
-
-	// It's okay to skip the mutex on this next line: g.isConnectedToNtrip can only be mutated by this
-	// goroutine itself
-	for g.isConnectedToNtrip && !g.isClosed {
+	for !g.isClosed {
 		select {
 		case <-g.cancelCtx.Done():
 			return
 		default:
 		}
 
+		// Despite this looking like we're getting the next message from the scanner, what we're
+		// really doing is just checking if the scanner is still connected to the mount point. If
+		// it's disconnected, we'll get an error, which is our signal to reconnect.
```

Comment:
The comment is a bit misleading. According to the [go rtcm package](https://pkg.go.dev/github.com/go-gnss/rtcm/rtcm3#Scanner) Scanner is used for parsing RTCM messages from a reader, which is what we are doing in scanner.NextMessage()

Reply:
If you really want to add a comment here, I would recommend that you say something similar to what the description of NextMessage() is saying

Reply:
How does this new version of the comment sound?

Also, thanks for pointing out that we can never have a non-nil message and non-nil error at the same time: that let me delete a paragraph of dead code, which let me remove the `msg` variable entirely, which helps clarify why there's a comment here.